File: hackerrank/day10/day10.js
'use strict';

const fs = require('fs');

const n = parseInt(fs.readFileSync(0, 'utf8').trim());

// Convertir a binario
const binario = n.toString(2);

let maximo = 0;
let contador = 0;

for (const bit of binario) {
    if (bit === '1') {
        contador++;
        maximo = Math.max(maximo, contador);
    } else {
        contador = 0;
    }
}

console.log(maximo);

File: hackerrank/day10/day10.py
#!/bin/python3

import math
import os
import random
import re
import sys

if __name__ == '__main__':
    n = int(input().strip())

    # Convertir a binario
    binario = bin(n)[2:]

    maximo = 0
    contador = 0

    for bit in binario:
        if bit == '1':
            contador += 1
            maximo = max(maximo, contador)
        else:
            contador = 0

    print(maximo)